support/builder/build_dep_populate.sh

```diff
@@ -0,0 +1,463 @@
+        echo "${failure}"
+    done
+    echo "########################################"
+    fi
+}
+
+putArtifacts() {
+    # shellcheck disable=SC2044
+    for file in $(find /hab/tmp/harts/ -name '*.hart'); do
+        if ! hab pkg upload --url "${bldr_url}" "${file}" --force; then
+            failed_uploads+=("${bldr_url}")
+        fi
+    done
+}
+
+genS3Config() {
+    echo ""
+    read -r -p "AWS Access Key ID: " access_key_id
+        export AWS_ACCESS_KEY_ID="${access_key_id}"
+    read -r -p "AWS Secret Access Key: " secret_access_key
+        export AWS_SECRET_ACCESS_KEY="${secret_access_key}"
+}
+
+genBldrCreds() {
+    echo ""
+    echo "Please provide a Builder Auth Token."
+    read -r -p "Builder Auth token: " bldr_auth_token
+        export HAB_AUTH_TOKEN="${bldr_auth_token}"
+    echo ""
+    echo "Builder auth token set to: ${HAB_AUTH_TOKEN}"
+    read -r -p "Would you like to use this auth token? [y/N]" response
+    if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+        return
+    else
+        genBldrCreds
+    fi
+}
+
+setBldrUrl() {
+    echo ""
+    echo "Package ingestion can be pointed towards public builder or"
+    echo "an on-prem builder instance."
+
+    if [[ -n ${HAB_BLDR_URL} ]]; then
+       echo ""
+       echo "Builder URL configured via ENVVAR detected."
+       echo ""
+       echo "HAB_BLDR_URL=${HAB_BLDR_URL}"
+       export bldr_url="${HAB_BLDR_URL}"
+    else
+        echo ""
+        read -r -p "Will you be uploading to public builder? [y/N]" response
+            if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                export bldr_url="https://bldr.habitat.sh"
+            else
+                echo ""
+                echo "Please provide the URL so your builder instance"
+                echo "Ex: https://bldr.habitat.sh or http://localhost"
+                read -r -p ": " response
+                    export bldr_url="${response}"
+                echo ""
+                echo "Your builder URL is now configured to ${bldr_url}"
+                read -r -p "Is this correct? [y/N]" response
+                    if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+                        return
+                    else
+                        setBldrUrl
+                    fi
+        fi
+    fi
+}
+
+credSelect() {
+    echo ""
+    read -r -p "Would you like to use these credentials? [y/N] " response
+    if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+        return
+    else
+        echo ""
+        read -r -p "Would you like to configure with custom credentials now? [y/N]" response
+        if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+            genS3Config
+        else
+            echo "Please reconfigure your AWS credentials and re-run s3migrate."
+            exit 1
+        fi
+    fi
+}
+
+credCheck(){
+    CREDS=( "${HOME}/.aws/credentials" "${HOME}/.aws/config" "/root/.aws/credentials" "/root/.aws/config")
```

Comment:
Same comments as before re all caps and creating a constant array for one time iteration